R/roclet-doxygen.R

```diff
@@ -0,0 +1,138 @@
+#' @title Replace a value for a given tag on file in memory
+#' @description Scan the lines and change the value for the named tag if one line has this tag, 
+#'    add a line at the end if no line has this tag and return a warning if several lines
+#'    matching the tag
+#' @param fileStrings A vector with each string containing a line of the file
+#' @param tag The tag to be searched for 
+#' @param newVal The new value for the tag
+#' @return The vector of strings with the new value
+#' @examples
+#' fakeFileStrings <- c("Hello = world","SURE\t= indeed","Hello = you")
+#' 
+#' expect <- warning(ReplaceTag(fakeFileStrings,"Hello","me"))
+#' 
+#' newFake <- ReplaceTag(fakeFileStrings,"SURE","me")
+#' expect <- equal(length(newFake), length(fakeFileStrings))
+#' expect <- equal(length(grep("SURE",newFake)), 1)
+#' expect <- equal(length(grep("me",newFake)), 1)
+#' 
+#' newFake <- ReplaceTag(fakeFileStrings,"Bouh","frightened?")
+#' expect <- equal(length(newFake), length(fakeFileStrings)+1)
+#' expect <- equal(length(grep("Bouh",newFake)), 1)
+#' expect <- equal(length(grep("frightened?",newFake)), 1)
+#' @keywords internal 
+ReplaceTag <- function(fileStrings,tag,newVal){
+    iLine <- grep(paste0("^",tag,"\\>"),fileStrings)
+    nLines <- length(iLine)
+    if(nLines == 0){
+        line <- paste0(tag,"\t= ",newVal)
+        iLine <- length(fileStrings)+1
+    }else if (nLines > 0){
+        line <- gsub("=.*",paste0("= ",newVal),fileStrings[iLine])
+        if(nLines >1){
+            warning(paste0("File has",nLines,"for key",tag,"check it up manually"))
+        }
+    }
+    fileStrings[iLine] <- line
+    return(fileStrings)
+}
+#' Prepares the R package structure for use with doxygen
+#' @description Makes a configuration file in inst/doxygen
+#'     and set a few options: 
+#'     \itemize{
+#'        \item{EXTRACT <- ALL = YES}
+#'        \item{INPUT = src/}
+#'        \item{OUTPUT <- DIRECTORY = inst/doxygen/}
+#'     }
+#' @param rootFolder The root of the R package
+#' @note Please note this implies the installation of doxygen on the system, 
+#'       accessible on the command line by `doxygen`
+#' @return NULL
+#' @examples 
+#' \dontrun{
+#' doxygen_init()
+#' }
+#' @keywords internal
+doxygen_init <- function(pkg_path,doxy_file){
+    doxDir <- dirname(doxy_file)
+    doxyFileName <- basename(doxy_file)
+
+    initFolder <- getwd()
+    if(pkg_path != "."){
+        setwd(pkg_path)
+    }
+    rootFileYes <- length(grep("DESCRIPTION",dir()))>0
+    # prepare the doxygen folder
+    if(!file.exists(doxDir)){
+        dir.create(doxDir,recursive=TRUE)
+    }
+    setwd(doxDir)
+
+    # prepare the doxygen configuration file
+    system(paste0("doxygen -g ",doxyFileName))
```

Comment:
I think it would be better to have a function called `doxygen()` that first checks that doxygen is available, and then wraps the call to `system()`. And as I think about it, that also needs a call to `shQuote()`. You might also want to call `normalizePath()` to ensure it works in the widest variety of settings.